### PR TITLE
private input with example, with cli refactor

### PR DESCRIPTION
**Pull request by rkdud007**
- private input support 
- private input example
- cli refactor: overall structure so that example entry point can be simply `hdp_run`
- cli refactor: `RUST_LOG` to control tracing level
- and some neat pick

### CODE REVIEW

File: examples/private-input-module/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    let hdp_run_config = HdpRunConfig {
+        dry_run_program_path: "../../build/contract_dry_run.json".into(),
+        sound_run_program_path: "../../build/hdp.json".into(),
+        pre_processor_output_file: pre_processor_output_file.into(),
+        processor_output_file: Some(output_file.into()),
+        cairo_pie_file: Some(cairo_pie_file.into()),
+        ..Default::default()
+    };
+
+    hdp_run::hdp_run(&hdp_run_config, tasks).await.unwrap();
```
Comment (Okm165) [inline examples/private-input-module/src/main.rs]: little bit weird syntax

File: hdp/src/preprocessor/compile/datalake/mod.rs [hunk 2]
```diff
@@ -14,42 +12,24 @@ impl Compilable for DatalakeCompute {
         &self,
         compile_config: &CompilerConfig,
     ) -> Result<CompilationResult, CompileError> {
-        info!("target task: {:#?}", self);
+        info!("Target task: {:#?}", self);
```
Comment (Okm165) [inline hdp/src/preprocessor/compile/datalake/mod.rs]:
```suggestion
        info!("target task: {:#?}", self);
```

File: hdp/src/preprocessor/module_registry.rs [hunk 3]
```diff
@@ -62,7 +65,7 @@ impl ModuleRegistry {
             program_hash.map(|program_hash| FieldElement::from_hex_be(&program_hash).unwrap());
         let module_inputs = module_inputs
             .into_iter()
-            .map(|input| FieldElement::from_hex_be(&input).unwrap())
+            .map(|input| ModuleInput::from_str(&input).unwrap())
```
Comment (Okm165) [inline hdp/src/preprocessor/module_registry.rs]: lets remove unwrap here
let map do the vec of results then collect it into result of vec and ? it

File: examples/private-input-module/src/main.rs (hunk 1 above)
Comment (rkdud007) [reply examples/private-input-module/src/main.rs]: you suggest `hdp_run(..)` directly?

Comment (Okm165) [reply examples/private-input-module/src/main.rs]: maybe smth like hdp::run() ?

Comment (Okm165) [reply examples/private-input-module/src/main.rs]: or hdp_run::run()

File: examples/private-input-module/.gitignore [hunk 4]
```diff
@@ -0,0 +1 @@
+*.zip
```
Comment (Okm165) [inline examples/private-input-module/.gitignore]: there is gitignore in this project already why is it needed here?

File: examples/private-input-module/src/main.rs (hunk 1 above)
Comment (rkdud007) [reply examples/private-input-module/src/main.rs:41]: i got idea from https://github.com/lambdaclass/cairo-vm/blob/93f1f54f38059af89850b1c06d4c901300f9b94d/vm/src/cairo_run.rs#L116 which i like than too general function name `run` i'll make export in upper module so user can call it as `hdp_run` not `hdp_run::hdp_run`

File: hdp/src/preprocessor/compile/datalake/mod.rs (hunk 2 above)
Comment (rkdud007) [reply hdp/src/preprocessor/compile/datalake/mod.rs]: changing all `info!` to lowercase

File: examples/private-input-module/src/main.rs (hunk 1 above)
Comment (rkdud007) [reply examples/private-input-module/src/main.rs]: `hdp::run`

**Review by Okm165 — APPROVED** [review]
LGTM

**Pull request merged**